impl/doc/error.md

```diff
@@ -32,6 +32,11 @@ often [`From` as well](crate::From).
    called `Backtrace`. Then it would return that field as the `backtrace`.
 3. One of the fields is annotated with `#[error(backtrace)]`. Then it would
    return that field as the `backtrace`.
+4. The source field is annotated with `#[error(backtrace)]`. Then it would
```

Comment:
copy paste error?
```suggestion
4. The source field is annotated with `#[error(forward)]`. Then it would
```

Reply:
No that's correct, `provide` is forwarded only when the source is annotated with `#[error(backtrace)]`.

That behaviour is unchanged in this PR.

The reasoning for keeping a separate attribute is to make source forwarding work on stable, since there's no good way of generating feature-gated code.

I also think this is somewhat unintuitive, but I haven't come up with a better idea.

Perhaps it could make sense to disallow `#[error(backtrace)]` on the source field if there's no `#[error(forward)]`?